### PR TITLE
Jetpack checklist: Inline tasks

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

* Move tasks that were spread across two files into the checklist.
* Move duration function into component
* Fix `complete` type error due to irregular checklist state
* Drop `flowRight` for easier typing
* Add `LocalizeProps`

#### Testing instructions

* Ensure the Jetpack Checklist works as expected (no functional changes).
* Different plan levels will include different checklist tasks, so try a few different plans.
* Visit `/plans/my-plan/SITE_SLUG` for a Jetpack site

Closes #33950

### CODE REVIEW

**Comment by sirreal** [comment]
Rebased to fix conflicts after #34050 and #33666.

Ready for review.

File: client/my-sites/plans/current-plan/jetpack-checklist/index.tsx [hunk 1]
```diff
@@ -172,47 +134,166 @@ class JetpackChecklist extends PureComponent< Props > {
...
+					{ isEnabled( 'jetpack/checklist/performance' ) && (
+						<Task
+							id="jetpack_site_accelerator"
+							completed={ this.isComplete( 'jetpack_site_accelerator' ) }
+							completedButtonText={ translate( 'Configure' ) }
+							completedTitle={ translate(
+								'Site accelerator is serving your images and static files through our global CDN.'
+							) }
+							description={ translate(
+								'Serve your images and static files through our global CDN and whatch your page load time drop.'
```
Comment (a8ci18n) [inline client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Serve your images and static files through our global CDN and watch your page load time drop.' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/remove/jetpack-checklist-constants)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (sirreal) [reply client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: This included a typo fixed by 2588fa7

File: client/my-sites/plans/current-plan/jetpack-checklist/index.tsx [hunk 2]
```diff
@@ -47,21 +37,20 @@ import withTrackingTool from 'lib/analytics/with-tracking-tool';
...
+		| {
+				id: string;
+				isCompleted: boolean;
+		  }[]
+		| undefined;
 	widgetCustomizerPaneUrl: URL | null;
 }
 
-class JetpackChecklist extends PureComponent< Props > {
+class JetpackChecklist extends PureComponent< Props & LocalizeProps > {
```
Comment (spen) [inline client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: Just for my own educational purposes here, `Props & LocalizeProps` will just combine the 2 interfaces right? And allow for a shape that's somewhat like:

```js
{
  ...someProps,
  ...someOtherProps,
}
```

Right?

Comment (sirreal) [reply client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: Yep, I think that's a valid way of expressing it 👍 

This is an [_intersection type_](https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types).

Please feel free to question or add feedback about anything TypeScript 🙌

File: client/my-sites/plans/current-plan/jetpack-checklist/index.tsx [hunk 3]
```diff
@@ -47,21 +37,20 @@ import withTrackingTool from 'lib/analytics/with-tracking-tool';
...
-}
-
 interface Props {
 	isPremium: boolean;
 	isProfessional: boolean;
 	isPaidPlan: boolean;
-	taskStatuses: ChecklistTaskState | undefined;
+	taskStatuses:
+		| {
+				id: string;
```
[the remark this answers is not in the document]
Comment (ockham) [reply client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: It's all prettified these days, so if we think anything's off, we'd need to tweak `prettier` :grimacing:

File: client/my-sites/plans/current-plan/jetpack-checklist/index.tsx [hunk 4]
```diff
@@ -172,47 +133,166 @@ class JetpackChecklist extends PureComponent< Props > {
 					progressText={ translate( 'Your Jetpack setup progress' ) }
 				>
 					<Task
-						{ ...JETPACK_CHECKLIST_TASK_PROTECT }
+						id="jetpack_task_protect"
+						title={ translate(
+							"We've automatically protected you from brute force login attacks."
```
Comment (ockham) [inline client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: Present tense here (I changed that in #33666, easy to miss when rebasing)

```suggestion
							"We're automatically protecting you from brute force login attacks."
```

File: client/my-sites/plans/current-plan/jetpack-checklist/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (spen) [reply client/my-sites/plans/current-plan/jetpack-checklist/index.tsx]: Ahh yes ofc! Thanks prettier! 😅

**Pull request merged**